File: Exercism Codes/Java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

public class Gigasecond {
    private LocalDateTime moment;
    public Gigasecond(LocalDate moment) {
        this.moment = moment.atStartOfDay();
    }

    public Gigasecond(LocalDateTime moment) {
        this.moment = moment;
    }

    public LocalDateTime getDateTime() {
        return moment.plus(1000000000, ChronoUnit.SECONDS);
    }
}

File: Exercism Codes/Kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.temporal.Temporal
import java.util.*

class Gigasecond {

    constructor(date : LocalDate) {
        this.date = date.atStartOfDay().plusSeconds(1000000000)
    }

    constructor(date : LocalDateTime) {
        this.date = date.plusSeconds(1000000000)
    }

    var date: LocalDateTime? = null
}
